### PR TITLE
Feature/school 664/main layout

### PR DESCRIPTION
**Pull request by mzavarka**

### CODE REVIEW

File: apps/schools/domains/common/components/Icons/IconWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import React, { CSSProperties, MouseEventHandler, SVGProps } from 'react'
+
+const SMALL_ICON_SIZE = 16
+const MEDIUM_ICON_SIZE = 20
+const LARGE_ICON_SIZE = 24
```
Comment (levil664) [inline apps/schools/domains/common/components/Icons/IconWrapper.tsx]: must be in consts/number

File: apps/schools/domains/common/components/Icons/IconWrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    color?: CSSProperties['color']
+    svgProps?: Pick<SVGProps<SVGSVGElement>, 'onClick'>
+    className?: string
+    id?: string
+    onClick?: MouseEventHandler<HTMLSpanElement>
+}
+
+type IconWrapperProps = Omit<IconProps, 'svgProps'> & {
+    icon: React.ReactNode
+}
```
Comment (levil664) [inline apps/schools/domains/common/components/Icons/IconWrapper.tsx]: must be in ./interfaces

File: apps/schools/domains/common/components/Icons/IconWrapper.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+                                                            onClick,
+                                                        }) => {
+    const iconSize = getIconSize(size)
+    const spanStyles: CSSProperties = {
+        color,
+        width: iconSize,
+        height: iconSize,
+        display: 'inline-flex',
+        fontSize: 'inherit',
+    }
```
Comment (levil664) [inline apps/schools/domains/common/components/Icons/IconWrapper.tsx]: must be in styles.module.scss

File: apps/schools/domains/common/components/Logo/index.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import Image from 'next/image'
+import React from 'react'
+import styles from './styles/styles.module.scss'
+import logo from '@public/icons/logo.svg'
+
+interface ILogoProps {
+    onClick?: (e: React.SyntheticEvent) => void
+    minified?: boolean
+    width?: number
+}
```
Comment (levil664) [inline apps/schools/domains/common/components/Logo/index.tsx]: must be in ./interfaces

File: apps/schools/domains/common/components/containers/BaseLayoutComponents/BaseLayout/index.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+import Menu from '../Menu'
+import {Icon} from "../Icon/Icon";
+import OrganizationSelect from "../OrganizationSelect";
+import UserProfile from "../UserProfile";
+
+const { Content, Sider } = Layout
+
+const LAYOUT_WIDTH = 255;
+const COLLAPSED_LAYOUT_WIDTH = 100;
```
Comment (levil664) [inline apps/schools/domains/common/components/containers/BaseLayoutComponents/BaseLayout/index.tsx]: must be in constants/number

File: apps/schools/domains/common/components/containers/BaseLayoutComponents/BaseLayout/index.tsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+interface IBaseLayoutProps extends Omit<PropsWithChildren, 'type'> {
+    style?: CSSProperties
+    className?: string
+    onLogoClick?: () => void
+}
```
Comment (levil664) [inline apps/schools/domains/common/components/containers/BaseLayoutComponents/BaseLayout/index.tsx]: must be in ./interfaces

File: apps/schools/domains/common/components/containers/BaseLayoutComponents/BaseLayoutContext.tsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+// import { ITopNotification, useTopNotificationsHook } from './TopNotifications'
+
+interface ILayoutContext {
+    isMobile?: boolean
+    shouldTableScroll?: boolean
+    isCollapsed?: boolean
+    toggleCollapsed: () => void
+    // addNotification?: (notification: ITopNotification) => void
+}
```
Comment (levil664) [inline apps/schools/domains/common/components/containers/BaseLayoutComponents/BaseLayoutContext.tsx]: ^^

File: apps/schools/domains/common/components/containers/BaseLayoutComponents/Icon/Icon.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import styles from "./styles/styles.module.scss";
+import classnames from "classnames";
+import {CustomLogo} from "../../../Logo";
+import {LeftOutlined, RightOutlined} from "@ant-design/icons";
+
+const LOGO_WIDTH = 50;
```
Comment (levil664) [inline apps/schools/domains/common/components/containers/BaseLayoutComponents/Icon/Icon.tsx]: must be in constants/number

File: apps/schools/domains/common/components/containers/BaseLayoutComponents/Icon/Icon.tsx [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+interface IconProps {
+    collapsed: boolean
+    setCollapsed: any
+}
```
Comment (levil664) [inline apps/schools/domains/common/components/containers/BaseLayoutComponents/Icon/Icon.tsx]: must be in interfaces folder

File: apps/schools/tsconfig.json [hunk 10]
```diff
@@ -27,7 +27,8 @@
             "@test/*": ["test/*"],
             "@test": ["test"]
         },
-        "incremental": true
+        //TODO: make it false for speed
+        "incremental": false
```
Comment (levil664) [inline apps/schools/tsconfig.json]: that's already false, isn't it?

File: apps/schools/domains/common/components/Icons/IconWrapper.tsx (hunk 1 above)
Comment (mzavarka) [reply apps/schools/domains/common/components/Icons/IconWrapper.tsx]: Can we delete IconWrapper? Is it useful?

File: apps/schools/tsconfig.json (hunk 10 above)
Comment (mzavarka) [reply apps/schools/tsconfig.json]: Yes, it was true. We should fix TODO

**Pull request merged**